### PR TITLE
Added metrics to etch VMModel::CompileSequential

### PR DESCRIPTION
**Pull request by evsmithx**
Added an optional list of metrics for etch model compilation, to match ML Model.

### CODE REVIEW

File: libs/ml/include/ml/model/model.hpp [hunk 1]
```diff
@@ -490,6 +508,11 @@ void Model<TensorType>::TrainImplementation(DataType &loss, SizeType n_rounds)
   loss = loss_;
 }
 
+template <typename TensorType>
+bool Model<TensorType>::DataLoaderIsSet()
+{
+  return dataloader_ptr_->Size() != 0;
```
Comment (baykaner) [inline libs/ml/include/ml/model/model.hpp:518]: is it possible for this to be called while dataloader_ptr_ is still a null_ptr?

File: libs/vm-modules/include/vm_modules/ml/model/model_estimator.hpp [hunk 2]
```diff
@@ -239,6 +253,7 @@ class ModelEstimator
     // Model
     DataType forward_pass_cost{0.0};
     DataType backward_pass_cost{0.0};
+    DataType metrics_cost{0.0};
```
Comment (baykaner) [inline libs/vm-modules/include/vm_modules/ml/model/model_estimator.hpp]: all of these can now be string initialisations (maybe coordinate with Jiri for the nicest way to do this for t he static constexpr functions above)

File: libs/ml/include/ml/model/model.hpp (hunk 1 above)
Comment (baykaner) [inline libs/ml/include/ml/model/model.hpp:518]: can dataloader_ptr_ ever be a null ptr?

File: libs/vm-modules/include/vm_modules/ml/model/model_estimator.hpp [hunk 3]
```diff
@@ -179,6 +185,14 @@ class ModelEstimator
   {
     return DataType(0.003333333333333);
   };
+  static constexpr DataType SCEL_FORWARD_IMPACT()
+  {
+    return DataType(0.003333333333333);
```
Comment (baykaner) [inline libs/vm-modules/include/vm_modules/ml/model/model_estimator.hpp]: Konstantinos' PR is in, so FixedPoint initialisations should all be strings

File: libs/vm-modules/tests/unit/ml/ml_model_tests.cpp [hunk 4]
```diff
@@ -768,4 +768,44 @@ TEST_F(VMModelTests, DISABLED_regressor_model_test)
...
+          label_shape[1] = 1000u64;
+          var data = Tensor(data_shape);
+          var label = Tensor(label_shape);
+
+          // set up model
+          var model = Model("sequential");
+          model.add("dense", 10u64, 10u64, "relu");
+          model.add("dense", 10u64, 10u64, "relu");
+          model.add("dense", 10u64, 1u64);
+          model.compile("mse", "adam", {"mse"});
```
Comment (baykaner) [inline libs/vm-modules/tests/unit/ml/ml_model_tests.cpp:791]: nice 👍 

can you also demonstrate using accuracy?

File: libs/vm-modules/src/ml/model/model_estimator.cpp [hunk 5]
```diff
@@ -216,20 +216,95 @@ ChargeAmount ModelEstimator::CompileSequential(Ptr<String> const &loss,
          COMPUTE_CHARGE_COST;
 }
 
+ChargeAmount ModelEstimator::CompileSequentialWithMetrics(
```
Comment (baykaner) [inline libs/vm-modules/src/ml/model/model_estimator.cpp:259]: you also need to add a couple of estimator tests (see vm-modules/test/unit/model/model_estimator); these should be sanity checks that just ensure the model estimator does what we expect it to over a range of input values

**Review by baykaner — COMMENTED** [review]
very nice; a few extra bits needed and then we can merge this in

File: libs/ml/include/ml/model/model.hpp (hunk 1 above)
Comment (evsmithx) [reply libs/ml/include/ml/model/model.hpp:518]: Good catch!

File: libs/vm-modules/include/vm_modules/ml/model/model_estimator.hpp (hunk 3 above)
Comment (evsmithx) [reply libs/vm-modules/include/vm_modules/ml/model/model_estimator.hpp]: Jiri has done these in his PR

**Review by vhavrylov-ncube — COMMENTED** [review]
Great job!

**Review by baykaner — APPROVED** [review]
